### PR TITLE
[quality] Update repository quality suite

### PR DESCRIPTION
**Pull request by cfrantz**
Refactor the "quality" suite for the OpenTitan repository.

TL;DR: `bazel test //quality/...`

1. Move all of the quality check/fix targets to //quality.
2. Refactor the quality check targets into bazel tests (rather than executables).  The motivation for this change is both ergonomic and aesthetic. One ought to be able to type:
   ```
   bazel test //quality/... [--test_output=streamed]
   ```
3. Update various CI scripts that used to `bazel run <some_check>` to instead `bazel test <some_check> --test_output=streamed`.
4. Update the bazel user-guide/cheat sheet to reflect the `test` invocation and new labels.

This change should not be committed until after https://github.com/lowRISC/misc-linters/pull/30.

### CODE REVIEW

**Comment by cfrantz** [comment]
Oh, irony:  The `quality` refactor failed the CI's quality stage.

**Review by timothytrippel — APPROVED** [review]

File: third_party/lint/repos.bzl [hunk 1]
```diff
@@ -3,23 +3,28 @@
...
+def lint_repos(lowrisc_lint = None):
     # We have a 'vendored' copy of the google_verible_verilog_syntax_py repo
     native.local_repository(
         name = "google_verible_verilog_syntax_py",
         path = "hw/ip/prim/util/vendor/google_verible_verilog_syntax_py",
     )
 
+    # FIXME(cfrantz): Name collision with another package loading `com_github_bazelbuild_buildtools`?
+    # We seem to always be getting an _old_ version of buildtools that doesn't have the
+    # `buildify_test` rule.  Temporary workaround.
```
Comment (drewmacrae) [inline third_party/lint/repos.bzl:17]: Pinning to a specific version is probably fine in the long run: Lets us control when we take on merge skew risk.

File: quality/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+clang_format_exclude = [
+    # Vendored source code dirs
+    "./**/vendor/**",
+    # Rust cargo build dirs
+    "./**/target/**",
+    # Directories used exclusively to store build artifacts are still copied into.
+    "./build-out/**",
+    "./build-bin/**",
+    # fusesoc build dir
```
Comment (drewmacrae) [inline quality/BUILD.bazel:84]: Nit: I think the docs also get built in `build`

File: third_party/lint/repos.bzl [hunk 3]
```diff
@@ -3,23 +3,28 @@
...
     http_archive(
-        name = "com_github_bazelbuild_buildtools",
-        strip_prefix = "buildtools-main",
-        url = "https://github.com/bazelbuild/buildtools/archive/main.zip",
+        name = "xcom_github_bazelbuild_buildtools",
```
Comment (drewmacrae) [inline third_party/lint/repos.bzl:19]: Rule 'xcom_github_bazelbuild_buildtools' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "e3bb0dc8b0274ea1aca75f1f8c0c835adbe589708ea89bf698069d0790701ea3"

**Review by drewmacrae — APPROVED** [review]
LGTM, Thanks!

File: third_party/lint/repos.bzl (hunk 3 above)
Comment (timothytrippel) [reply third_party/lint/repos.bzl:19]: I should really find a way to make missing SHA's blocking in CI. I filed #15110 to track.

Comment (cfrantz) [reply third_party/lint/repos.bzl]: I'd always intended to fix this before committing and its now fixed.

File: third_party/lint/repos.bzl (hunk 1 above)
Comment (cfrantz) [reply third_party/lint/repos.bzl]: Its a best practice... thus spake the TL :)

File: quality/BUILD.bazel (hunk 2 above)
Comment (cfrantz) [reply quality/BUILD.bazel:84]: I'm not sure what the suggestion is here.  This was copied directly from the prior incarnation in the root BUILD file.

These excludes are slated for deletion once we get the CI doing everything through bazel.

**Pull request merged**